pkg/meta/meta.go

```diff
@@ -1560,7 +1576,50 @@ func (m *Meta) GetHistoryDDLJobsIterator(startJobID int64) (LastJobIterator, err
 
 // HLastJobIterator is the iterator for gets the latest history.
 type HLastJobIterator struct {
-	iter *structure.ReverseHashIterator
+	iter       *structure.ReverseHashIterator
+	schemaIDs  set.Int64Set
+	tableNames set.StringSet
+}
+
+// extractSchemaIDAndTableName extract schema_id and table_id from encoded Job structure
+// Note, here we strongly rely on the order of fields in marshalled string, just like checkSubstringsInOrder
+func extractSchemaIDAndTableName(s string) (int64, string, error) {
+	pos := strings.Index(s, `"schema_id":`)
+	if pos == -1 {
+		return 0, "", fmt.Errorf("schema_id not found in model.Job json")
+	}
+
+	start := pos + len(`"schema_id":`)
+	substr := s[start:]
+	end := strings.Index(substr, ",")
+	if end == -1 {
+		end = len(substr)
+	}
+
+	schemaIDStr := strings.TrimSpace(substr[:end])
+	schemaID, err := strconv.ParseInt(schemaIDStr, 10, 64)
+	if err != nil {
+		return 0, "", fmt.Errorf("failed to parse table_id: %v", err)
+	}
+
+	pos = strings.Index(substr, `"table_name":`)
+	if pos == -1 {
+		return 0, "", fmt.Errorf("table_name not found in model.Job json")
+	}
+
+	start = pos + len(`"table_name":`)
+	substr = substr[start:]
+	end = strings.Index(substr, ",")
+	if end == -1 {
+		end = len(substr)
+	}
+
+	tableName := strings.TrimSpace(substr[:end])
+	if strings.HasPrefix(tableName, `"`) && strings.HasSuffix(tableName, `"`) {
```

Comment:
Does it always have `"` because it's a JSON string? we don't need to check it.

Reply:
I'll check it and add some tests too.